source/common/router/rds_impl.cc

```diff
@@ -48,7 +48,8 @@ StaticRouteConfigProviderImpl::StaticRouteConfigProviderImpl(
     Server::Configuration::FactoryContext& factory_context,
     RouteConfigProviderManagerImpl& route_config_provider_manager)
     : config_(new ConfigImpl(config, factory_context.getServerFactoryContext(),
-                             factory_context.messageValidationVisitor(), true)),
+                             factory_context.messageValidationContext().staticValidationVisitor(),
```

Comment:
I don't think this is right. You can have a static inline route in a dynamic `Listener`, for example, and we should be applying dynamic validation logic.

Reply:
revert it.
In the long run, I will probably make Listener validator passed down to RDS and its children object. (so that RDS doesn't depend on Listener).